hypervisor/arch/x86/cpuid.c

```diff
@@ -158,9 +173,36 @@ static void init_vcpuid_entry(__unused struct vm *vm,
 		break;
 
 	default:
-		cpuid_subleaf(leaf, subleaf,
-				&entry->eax, &entry->ebx,
-				&entry->ecx, &entry->edx);
+		if (leaf >= 0x80000000U) {
+			/* unsupported extended cpuid level */
+			if (leaf > boot_cpu_data.extended_cpuid_level) {
+				entry->eax = 0U;
```

Comment:
The hardware must support leaf 0x15. The only one presented to guest (but not in hardware) is leaf 0x16 but it is already covered before this code. Do we need this code?

Reply:
remove dead code to keep small